app/models/input_file.rb

```diff
@@ -25,10 +25,10 @@ def s3_source_check
     end
   end
 
-  after_validation(on: :create) do
+  def set_presigned_url_for_local_upload
```

Comment:
can you just make this an  after_save hook?

Reply:
yes, thanks